### PR TITLE
Add changelog entries

### PR DESCRIPTION
**Pull request by rakanalh**

### CODE REVIEW

File: docs/changelog.rst [hunk 1]
```diff
@@ -1,8 +1,37 @@
 =========
 Changelog
 =========
-
 * :release:`0.100.3-rc5 <2019-05-08>`
+* :bug:`4080` Fix "Unexpected signature format" issue in matrix transport
+* :feature:`4043` Update raiden-contracts to 0.19.0 with Görli Testnet support
+* :bug:`4041` Fix V20 -> V21 migration chain-state transformation to canonical identifiers
+* :bug:`4024` Fix clearing a channel state if and only if all unlocks are done by channel participants
+* :feature:`4002` Nightly builds are now deployed to Docker Hub
```
Comment (LefterisJP) [inline docs/changelog.rst]: This is an infrastructure change, not related to the raiden client software itself.

File: docs/changelog.rst [hunk 2]
```diff
@@ -1,8 +1,37 @@
...
+* :bug:`3960` Fix KeyError for canonical identifier during Raiden restart
+* :bug:`3874` Fix invalidation of a batch-unlock transaction in case a similar transaction was already sent/mined by channel partner
+* :bug:`3856` Handle pruned blocks during settle/unlock when requesting on-chain state, use latest if block is pruned
+* :bug:`3850` Fix IOU request from PFS caused by not sending a complete payload as the endpoint requires
+* :feature:`3482` Provide ARM builds during CI deployments
+* :bug:`3832` Fix Raiden startup when a previous run aborted during migrations
+* :feature:`3798` Mypy issues are fixed on the Raiden codebase
```
Comment (LefterisJP) [inline docs/changelog.rst]: Mypy does not concern the user

File: docs/changelog.rst [hunk 3]
```diff
@@ -1,8 +1,37 @@
...
+* :bug:`3790` Fix V19 -> V20 migration bug which did not initialize the onchain locksroots with the correct values
+* :bug:`3765` Fix matrix transport's global room initialization which lead messages to be sent to the wrong room
+* :bug:`3757` Fix a bug in PFS random selection using a confirmed block
+* :bug:`3743` Fix token network to payment networking mapping
+* :feature:`3738` Raiden codebase is now code-formatted using Black
```
Comment (LefterisJP) [inline docs/changelog.rst]: Users don't care about this

File: docs/changelog.rst [hunk 4]
```diff
@@ -1,8 +1,37 @@
...
+* :bug:`3732` Fix a startup issue where PFS required the user-deposit contract address in development mode
+* :bug:`3710` Fix Migration V19 -> V20 which caused Raiden to crash when searching for ReceiveContractChannelNew
+* :feature:`3697` Make sure a token implements the ERC20 interface when registering a new token network. In this case, totalSupply function existence is implemented
+* :bug:`3681` Fix an issue in migration V18 -> V19 which caused Raiden to crash
+* :bug:`3687` Fix startup initialization issue which caused Raiden to crash on private chains
+* :feature:`3677` Broke the raiden/ui module down into smaller testable components
```
Comment (LefterisJP) [inline docs/changelog.rst]: Users don't care about refactorings

File: docs/changelog.rst [hunk 5]
```diff
@@ -1,8 +1,37 @@
...
+* :bug:`4080` Fix "Unexpected signature format" issue in matrix transport
+* :feature:`4043` Update raiden-contracts to 0.19.0 with Görli Testnet support
+* :bug:`4041` Fix V20 -> V21 migration chain-state transformation to canonical identifiers
+* :bug:`4024` Fix clearing a channel state if and only if all unlocks are done by channel participants
+* :feature:`4002` Nightly builds are now deployed to Docker Hub
+* :bug:`3960` Fix KeyError for canonical identifier during Raiden restart
+* :bug:`3874` Fix invalidation of a batch-unlock transaction in case a similar transaction was already sent/mined by channel partner
+* :bug:`3856` Handle pruned blocks during settle/unlock when requesting on-chain state, use latest if block is pruned
+* :bug:`3850` Fix IOU request from PFS caused by not sending a complete payload as the endpoint requires
+* :feature:`3482` Provide ARM builds during CI deployments
```
Comment (LefterisJP) [inline docs/changelog.rst]: This is not something related to the changes contained in the new version, so should not be in the changelog.

File: docs/changelog.rst [hunk 6]
```diff
@@ -1,8 +1,37 @@
 =========
 Changelog
 =========
-
 * :release:`0.100.3-rc5 <2019-05-08>`
```
Comment (LefterisJP) [inline docs/changelog.rst]: Finally a comment on the too many bugs. In the past bugs between RCs were not added in the changelog as they are not supposed to be user facing bugs as users are not supposed to use RCs. If we do the changelog blows up like it does right here and imo loses its meaning.

Comment (rakanalh) [reply docs/changelog.rst]: I removed all bug fixes which were fixed in between releases.

File: docs/changelog.rst [hunk 7]
```diff
@@ -1,8 +1,20 @@
...
+* :feature:`4043` Update raiden-contracts to 0.19.0 with Görli Testnet support
+* :bug:`4024` Fix clearing a channel state if and only if all unlocks are done by channel participants
```
Comment (LefterisJP) [inline docs/changelog.rst]: nitpick: if and only if -> iff

**Pull request merged**